doc/dev/adr/0022-codegen-err-headers.md

```diff
@@ -0,0 +1,258 @@
+
+The previous section described the use of adjunct types `ErrorCode` and `ErrorInfo` within an `Object` co-typed `Result`.
+These adjunct types may also be used within an `Object` co-typed `Error`.
+
+The following DTDL model is similar to the above enhanced model, but it adds a `Field` with co-type `ErrorResult` to the `Object/Result`, similar to the example in ADR 19.
+Also, the `Field`s with new adjunct types have been relocated from the `Object/Result` to the `Object/Error` that is the "schema" of the `ErrorReslt`.
+
+```json
+{
+  "@context": [ "dtmi:dtdl:context;4", "dtmi:dtdl:extension:mqtt;4" ],
+  "@id": "dtmi:com:example:CounterCollection;1",
+  "@type": [ "Interface", "Mqtt" ],
+  "commandTopic": "rpc/command-samples/{executorId}/{commandName}",
+  "payloadFormat": "Avro/1.11.0",
+  "contents": [
+    {
+      "@type": "Command",
+      "name": "increment",
+      "request": {
+        "name": "counterName",
+        "schema": "string"
+      },
+      "response": {
+        "name": "incrementResponse",
+        "schema": {
+          "@type": [ "Object", "Result" ],
+          "fields": [
+            {
+              "@type": [ "Field", "NormalResult" ],
+              "name": "counterValue",
+              "schema": "integer"
+            },
+            {
+              "@type": [ "Field", "ErrorResult" ],
+              "name": "incrementError",
+              "schema": "dtmi:com:example:CounterCollection:CounterError;1"
+            }
+          ]
+        }
+      }
+    }
+  ],
+  "schemas": [
+    {
+      "@id": "dtmi:com:example:CounterCollection:CounterError;1",
+      "@type": [ "Object", "Error" ],
+      "fields": [
+        {
+          "@type": [ "Field", "ErrorMessage" ],
+          "name": "explanation",
+          "schema": "string"
+        },
+        {
+          "@type": [ "Field", "ErrorCode" ],
+          "name": "appErrCode",
+          "schema": "dtmi:com:example:CounterCollection:AppErrCode;1"
+        },
+        {
+          "@type": [ "Field", "ErrorInfo" ],
+          "name": "appErrPayload",
+          "schema": {
+            "@type": "Array",
+            "elementSchema": "string"
+          }
+        }
+      ]
+    },
+    {
+      "@id": "dtmi:com:example:CounterCollection:AppErrCode;1",
+      "@type": "Enum",
+      "valueSchema": "string",
+      "enumValues": [
+        {
+          "name": "success",
+          "enumValue": "succès"
+        },
+        {
+          "name": "failure",
+          "enumValue": "échec"
+        }
+      ]
+    }
+  ]
+}
+```
+
+As described in ADR 19, the error processing path may differ from the normal processing path.
+For instance, in C#, the server indicates an error by throwing an exception type that is generated from the `Object` co-typed `Error`.
+The use of an exception prevents the server code from calling the `WithAWithApplicationError` method because the `ExtendedResponse` is instantiated within the generated exception handler.
```

Comment:
```suggestion
The use of an exception prevents the server code from calling the `WithApplicationError` method because the `ExtendedResponse` is instantiated within the generated exception handler.
```